### PR TITLE
Route Group Fundamental Sample

### PR DESCRIPTION
**Pull request by fiyazbinhasan**
Fixes https://github.com/dotnet/AspNetCore.Docs/issues/26819

@tdykstra please review, I was unable to create a relative path using linkGenerator. 

I tried with,

`linkGenerator.GetUriByRouteValues(context, null, new {id = todo.Id})`, but it always returns null. Do you have any idea or tips on generating a link using the technique mentioned in the blog? 

> Instead of using relative addresses for the Location header in the 201 Created result, it’s also possible to use GroupRouteBuilder.GroupPrefix to construct a root-relative address or use [GetPathByRouteValues or GetUriByRouteValues](https://docs.microsoft.com/dotnet/api/microsoft.aspnetcore.routing.linkgeneratorroutevaluesaddressextensions) with a routeName for even more options.

### CODE REVIEW

**Comment by fiyazbinhasan** [comment]
@halter73 could you shed some light on this? I found an open issue that is quite close to this,

https://github.com/dotnet/aspnetcore/issues/41430

> Fixes #26819
> 
> @tdykstra please review, I was unable to create a relative path using linkGenerator.
> 
> I tried with,
> 
> `linkGenerator.GetUriByRouteValues(context, null, new {id = todo.Id})`, but it always returns null. Do you have any idea or tips on generating a link using the technique mentioned in the blog?
> 
> > Instead of using relative addresses for the Location header in the 201 Created result, it’s also possible to use GroupRouteBuilder.GroupPrefix to construct a root-relative address or use [GetPathByRouteValues or GetUriByRouteValues](https://docs.microsoft.com/dotnet/api/microsoft.aspnetcore.routing.linkgeneratorroutevaluesaddressextensions) with a routeName for even more options.

**Comment by tdykstra** [comment]
>Do you have any idea or tips on generating a link using the technique mentioned in the blog?

I do not. The first suggestion mentioned, the GroupPrefix property, doesn't even exist so far as I can tell.

File: aspnetcore/fundamentals/minimal-apis/7.0-samples/todo-group2/Program.cs
Comment (tdykstra) [inline aspnetcore/fundamentals/minimal-apis/7.0-samples/todo-group2/Program.cs:25]:
```suggestion
    .WithTags("Public Todo Endpoints");
```

**Review by tdykstra — APPROVED** [review]
Thank you, this looks good.  We can wait on merging it to see if you get an answer to your question about link generation for the Location header.

**Comment by fiyazbinhasan** [comment]
@tdykstra I figured out how to generate links using the `linkGenerator`. In that case, we have to modify `MapGroup` int the `Program.cs` with a pattern as follows,

```
app.MapGroup("/{accessibility}/todos")
    .MapTodosApi()
    .WithTags("Todo Endpoints");
```

Then have to modify the endpoints as follows,

```
public static RouteGroupBuilder MapTodosApi(this RouteGroupBuilder group)
{
    group.MapGet("/{id}", GetTodo).WithName("GetTodo");
    group.MapPost("/", CreateTodo);

    return group;
}

// get todo by id
public static async Task<Results<Ok<Todo>, NotFound>> GetTodo(int id, TodoGroupDbContext database)
{
    var todo = await database.Todos.FindAsync(id);

    if (todo is null)
        return TypedResults.NotFound();
    
    return TypedResults.Ok(todo);
}

// create todo
public static async Task<Created<Todo>> CreateTodo(Todo todo, TodoGroupDbContext database, HttpContext context, LinkGenerator linkGenerator)
{
    await database.Todos.AddAsync(todo);
    await database.SaveChangesAsync();

    var link = linkGenerator.GetUriByRouteValues(context, "GetTodo", new { id = todo.Id });
    
    return TypedResults.Created($"{link}", todo);
}
```

Maybe you have to use some preprocessor operators to show both use cases in the sample. I can modify the sample as per your requirements. Let me know. Thanks

**Comment by fiyazbinhasan** [comment]
You have to pass in the `{accessiblity}` route param,
 
image

**Comment by tdykstra** [comment]
>show both use cases in the sample

I see one use case in this code, what are the two use cases?

**Comment by fiyazbinhasan** [comment]
These are very different configurations,

```
app.MapGroup("/{accessibility}/todos")
    .MapTodosApi()
    .WithTags("Todo Endpoints");

group.MapGet("/{id}", GetTodo).WithName("GetTodo");
```

With the following approach, you **can't** assign any name to your group. It will throw you an exception saying the name should be globally unique, hence `linkGenerator` won't be able to generate any links.

```
app.MapGroup("/public/todos")
    .MapTodosApi()
    .WithTags("Public Todo Endpoints");

app.MapGroup("/private/todos")
    .MapTodosApi()
    .RequireAuthorization();
```

I leave it to you. I think you can explain it better or modify the sample appropriately for the doc

**Comment by tdykstra** [comment]
Let's go with your new code then,

**Comment by fiyazbinhasan** [comment]
@tdykstra It would be best if someone from the product team could intervene here. Maybe I'm missing some crucial configuration in order to make it work

> These are very different configurations,
> 
> ```
> app.MapGroup("/{accessibility}/todos")
>     .MapTodosApi()
>     .WithTags("Todo Endpoints");
> 
> group.MapGet("/{id}", GetTodo).WithName("GetTodo");
> ```
> 
> With the following approach, you **can't** assign any name to your group. It will throw you an exception saying the name should be globally unique, hence `linkGenerator` won't be able to generate any links.
> 
> ```
> app.MapGroup("/public/todos")
>     .MapTodosApi()
>     .WithTags("Public Todo Endpoints");
> 
> app.MapGroup("/private/todos")
>     .MapTodosApi()
>     .RequireAuthorization();
> ```

**Pull request closed without merging**
